test/extensions/filters/http/custom_response/utility.h

```diff
@@ -0,0 +1,206 @@
+                    "@type": type.googleapis.com/envoy.type.matcher.v3.HttpResponseStatusCodeMatchInput
+                value_match:
+                  exact: "503"
+            - single_predicate:
+                input:
+                  name: "504_response"
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.type.matcher.v3.HttpResponseStatusCodeMatchInput
+                value_match:
+                  exact: "504"
+        on_match:
+          action:
+            name: action
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.custom_response.v3.RedirectPolicy
+              status_code: 299
+              host: "https://foo.example"
+              path: "/gateway_error"
+              response_headers_to_add:
+              - header:
+                  key: "foo2"
+                  value: "x-bar2"
+      - predicate:
+          single_predicate:
+            input:
+              name: "500_response"
+              typed_config:
+                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpResponseStatusCodeMatchInput
+            value_match:
+              exact: "500"
+        on_match:
+          action:
+            name: action
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.custom_response.v3.RedirectPolicy
+              status_code: 292
+              host: "https://some.other.host"
+              path: "/internal_server_error"
+              response_headers_to_add:
+              - header:
+                  key: "foo3"
+                  value: "x-bar3"
+              request_headers_to_add:
+              - header:
+                  key: "cer-only"
+                keep_empty_value: true
+      - predicate:
+          single_predicate:
+            input:
+              name: "520_response"
+              typed_config:
+                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpResponseStatusCodeMatchInput
+            value_match:
+              exact: "520"
+        on_match:
+          action:
+            name: action
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.custom_response.v3.RedirectPolicy
+              host: "https://global/storage"
+              path: "/internal_server_error"
+              response_headers_to_add:
+              - header:
+                  key: "foo3"
+                  value: "x-bar3"
+##            modify_request_headers_action:
+##              name: modify-request-headers-action
+##              typed_config:
+##                "@type": type.googleapis.com/google.protobuf.Struct
+  )EOF";
+
+// Simulate filters that send local reply during either encode or decode based
+// on route specific config.
+class LocalReplyDuringDecodeIfNotCER : public Http::PassThroughFilter {
+public:
+  constexpr static char name[] = "local-reply-during-decode-if-not-cer";
```

Comment:
Why are these strings (also in `LocalReplyDuringEncodeIfNotCER` and `LocalReplyDuringDecode`) needed?
If it's for a unique id for the filter, it should be discouraged (per #21763).

Reply:
The names are required for specifying the filter. See for example `RouteSpecificDecodeLocalReplyBeforeRedirectedCER`  in custom_response_integration_test.cc

Reply:
Added typed config.